### PR TITLE
Use UIImagePickerController to determine existence of camera

### PR DESCRIPTION
**Pull request by mollidor**
AVCaptureDevice returns no devices if the authorization mode is
AVAuthorizationStatusDenied or AVAuthorizationStatusRestricted.
Thus the camera appeared to be non-existent if camera usage has been
denied or is restricted.
+[UIImagePickerController availableCaptureModesForCameraDevice:]
includes UIImagePickerControllerCameraCaptureModeVideo regardless of
the authorization status.

### CODE REVIEW

**Comment by mollidor** [comment]
@shagedorn please review. Will create PR to upstream once happy.

**Comment by mollidor** [comment]
commit message is confusing. Should read
> non-existent if camera usage is restricted.

It did return YES if denied before as well.

File: Classes/ios/Scanners/MTBBarcodeScanner.h [hunk 1]
```diff
@@ -139,9 +139,10 @@ typedef NS_ENUM(NSUInteger, MTBTorchMode) {
                                 previewView:(UIView *)previewView;
 
 /**
- *  Returns whether any camera exists in this device.
+ *  Returns whether any camera exists in this device regardless of camera permissions.
+ *  See scanningIsProhibited to check whehter access is denied.
```
Comment (mollidor) [inline Classes/ios/Scanners/MTBBarcodeScanner.h:143]: denied or restricted?

File: Classes/ios/Scanners/MTBBarcodeScanner.m [hunk 2]
```diff
@@ -174,21 +174,28 @@ - (void)dealloc {
...
+            return [self isVideoSupportedForCameraDevice:UIImagePickerControllerCameraDeviceFront];
     }
 
+    NSAssert(NO, @"Invalid camera type: %lu", (unsigned long)camera);
+
     return NO;
 }
 
++ (BOOL)isVideoSupportedForCameraDevice:(UIImagePickerControllerCameraDevice)device {
+    // returns YES even if camera access is AVAuthorizationStatusDenied or AVAuthorizationStatusRestricted
```
Comment (mollidor) [inline Classes/ios/Scanners/MTBBarcodeScanner.m:195]: should I just mention AVAuthorizationStatusRestricted here as it is the difference to AVCaptureDevice?

**Comment by shagedorn** [comment]
Is this issue related? https://github.com/mikebuss/MTBBarcodeScanner/issues/86

**Comment by mollidor** [comment]
Dang! Good catch. It definitely is. Stupid me.

**Comment by mollidor** [comment]
Will look into it once again and maybe the solution is to simply point out in the header that it returns `NO` if camera access is restricted (and provide a convenient method to check if restricted, maybe...).
What do you think?

**Comment by shagedorn** [comment]
Sounds good to me.

**Pull request closed without merging**